service/src/main/java/com/colatina/turmaformacao/tratofeito/service/repositorio/OfertaRepositorio.java

```diff
@@ -12,6 +12,15 @@
 @Repository
 public interface OfertaRepositorio extends JpaRepository<Oferta, Long> {
 
+    @Query("from Oferta o join o.itensOfertados itens where " +
+            "(o.item.id = :idItem or o.item.id IN :idItensOfertados or :idItem IN itens.id) " +
+            " and o.id <> :idOferta " +
+            "and o.situacao.id = :idSituacao" )
+    public List<Oferta> obterOfertasComItemAlvoTrocado(@Param("idItem") Long idItem,
```

Comment:
Operações de negações são muito custosas em banco. Colocar o `and o.id <> :idOferta` no final da query, assim ele só será executado caso todas as condições anteriores sejam verdadeira.
Reveja a ordem dos `ands`.

Reply:
feito